### PR TITLE
STM32H7: Fix wrong address for DTCM in linker script

### PR DESCRIPTION
**Pull request by multiplemonomials**
### Summary of changes 

I noticed that after #326, compilation is failing for STM32H7 targets!  This turns out to be because:
- The STM32H7 linker scripts use `NVIC_RAM_VECTOR_ADDRESS` to find the start of DTCM
- Most of the cmsis_nvic.h files define `NVIC_RAM_VECTOR_ADDRESS` based on `MBED_RAM_START`.  This was not really the correct way to do it, as on MCUs with multiple RAM banks, it's hard to know what ram bank MBED_RAM_START actually refers to
- MBED_RAM_START now points to SRAM_D2 rather than DTCM.
- This causes the linker script to try and store the crash data RAM in SRAM_D2 which is already used for the Ethernet stuff (which I actually plan to change soon, but oh well)

To fix this, I updated the cmsis_nvic.h files to remove the no-longer-needed RAM definitions and to define `NVIC_RAM_VECTOR_ADDRESS` in terms of the correct `MBED_RAM_BANK_SRAM_DTC_START` constant.

#### Impact of changes 
Compilation fixed for STM32H743, and possibly other MCUs in this family.

### Documentation 

None

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [X] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [X] Tests / results supplied as part of this PR
   
STM32H743 compiles again!
----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: platform/source/mbed_board.c [hunk 1]
```diff
@@ -28,8 +28,9 @@ WEAK MBED_NORETURN void mbed_die(void)
 #if !defined(TARGET_EFM32)
     core_util_critical_section_enter();
 #endif
-    gpio_t led_err;
+
 #ifdef LED1
+	gpio_t led_err;
```
Comment (multiplemonomials) [inline platform/source/mbed_board.c:33]: Fixing compiler warning if LED1 is not defined

File: tools/cmake/CheckPythonPackage.cmake [hunk 2]
```diff
@@ -27,7 +27,9 @@ function(check_python_package PACKAGENAME OUTPUT_VAR)
         set(PY_INTERP_FOR_${OUTPUT_VAR} ${Python3_EXECUTABLE} CACHE INTERNAL "The python interpreter used to run the ${OUTPUT_VAR} check" FORCE)
 
         execute_process(
-            COMMAND ${Python3_EXECUTABLE} -c "import ${PACKAGENAME}" RESULT_VARIABLE PACKAGECHECK_RESULT
+            COMMAND ${Python3_EXECUTABLE} -c "import ${PACKAGENAME}"
+            RESULT_VARIABLE PACKAGECHECK_RESULT
+            ERROR_QUIET
```
Comment (multiplemonomials) [inline tools/cmake/CheckPythonPackage.cmake:32]: Adding ERROR_QUIET to suppress the error message if the module is not found

File: tools/cmake/mbed_set_linker_script.cmake [hunk 3]
```diff
@@ -29,6 +29,11 @@ function(mbed_setup_linker_script mbed_os_target mbed_baremetal_target target_de
 
     # Find the path to the desired linker script
     # (the property should be set on both the OS and baremetal targets in a sane world)
+    get_property(RAW_LINKER_SCRIPT_PATHS_SET TARGET ${mbed_baremetal_target} PROPERTY INTERFACE_MBED_LINKER_SCRIPT SET)
+    if(NOT RAW_LINKER_SCRIPT_PATHS_SET)
+        message(FATAL_ERROR "No linker script has been set for the Mbed target. Ensure that code is calling mbed_set_linker_script() for the mbed-<your-board-name> target or one of its parents")
```
Comment (multiplemonomials) [inline tools/cmake/mbed_set_linker_script.cmake:34]: Ran into this while adding a custom target and the error message that it printed was not helpful.  Adding a better error message.

**Review by JohnK1987 — APPROVED** [review]

**Comment by wdx04** [comment]
I have a application that uses DMA for reading/writing SD cards in FAT filesystem. 
It works with the old linker script because it doesn't make use of DTCM.
But the new STM32H7 linker scripts seem to use DTCM aggressively, I can see 99.5% SRAM_DTC usage in the build output.
And my application fails with the new linker script (RX Overrun error), because the FATFilesystem buffer is located in DTCM.

**Comment by wdx04** [comment]
> I have a application that uses DMA for reading/writing SD cards in FAT filesystem. It works with the old linker script because it doesn't make use of DTCM. But the new STM32H7 linker scripts seem to use DTCM aggressively, I can see 99.5% SRAM_DTC usage in the build output. And my application fails with the new linker script (RX Overrun error), because the FATFilesystem buffer is located in DTCM.

Solved by removing the MBED_SPLIT_HEAP macro in my custom_targets.json5：

```json
"macros_remove": ["MBED_SPLIT_HEAP"]
```

**Comment by JojoS62** [comment]
I had the same issue, removing split heap fixed it. So removing this macro means that the additional heap_0 is not used as heap?

**Comment by multiplemonomials** [comment]
Update: this is fixed in master branch now!  We disabled DTCM for the heap

**Pull request merged**